include/circt/Dialect/FIRRTL/InnerSymbolTable.h

```diff
@@ -22,6 +22,56 @@
 namespace circt {
 namespace firrtl {
 
+/// The target of an inner symbol, the entity the symbol is a handle for.
+class InnerSymTarget {
```

Comment:
High level: this makes a lot of sense. `AnnoTarget` proved to be immensely useful. This is the inner sym analog.

Reply:
Yes, it kinda hurt to not use `AnnoTarget` since it represents rather similar things, but ultimately wanted something clearly separate at least for now and so not to have issues if one or the other wants to change in some minor but incompatible way.  Unifying them a bit might be a good idea in the future.  Compared to `AnnoTarget`, this is a bit simpler for now but may want to be more like `AnnoTarget` if we want per-target-type methods, etc..